src/core/config/srp_client.h

```diff
@@ -70,6 +70,19 @@
 #define OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_MODE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_ANYCAST_SERVER_PORT
+ *
+ * Specifies the default port number to use to connect to a SRP server selected using anycast address from a related
+ * service entry in Thread Network Data.
+ *
+ * This config is applicable only when `OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_API_ENABLE` is enabled.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_ANYCAST_SERVER_PORT
+#define OPENTHREAD_CONFIG_SRP_CLIENT_AUTO_START_DEFAULT_ANYCAST_SERVER_PORT 53
```

Comment:
I think it's fine to use same port (i.e. 53) for both SRP server and DNS server.

Reply:
Why? `53` is not allocated for SRP.

Reply:
For anycast model, we want to assume/use a fixed and pre-known port number (which we need to define and inlcude in spec).

We can use port number 53 but it may add some complications in the servers side implementation (e.g., may need to share the same socket to receive for `Srp::Server` and `Dnssd::Server`). So I persoanlly prefer if we can define and use a different port number. 

So for now, I added it as a config option.

Reply:
> We can use port number 53 but it may add some inconveniences in the servers side implementation (e.g., may need to share the same socket to receive for `Srp::Server` and `Dnssd::Server`). So I persoanlly prefer if we can define and use a different port number.

It's not that difficult. We can create one server UDP socket and dispatch the message according to the DNS type.

Reply:
> We can use port number 53 but it may add some inconveniences in the servers side implementation...

@abtink I don't think so. I think we will be violating the [IANA registration rules](https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml) if we use `53` for SRP server. As we know, the SRP protocol is a subset of the [RFC2136 DNS UPDATE](https://tools.ietf.org/html/rfc2136) (which uses port `53`) with additional constrains on the DNS records. A server which provides both standard DNS UPDATE and SRP service will not be able to tell whether a DNS message is a SRP message or just a normal DNS UPDATE message.

Reply:
@wgtdkp that may be a good argument for why it'd be better to define and use a different fixed port number for SRP.  I don't know if the idea/intention of reusing same port may be that the message is first given to SRP server to process and if it cannot process it, then the message is passed to DNS server (but there may be situations that such a model may not work). 

I also prefer using different port numbers (it is a cleaner model, the intended protocol being used is clear from the port number, and we can avoid the implementation inconveniences like sharing the same sockets, etc).

Reply:
[draft-ietf-dnssd-srp-09 Section 2.3.2](https://tools.ietf.org/html/draft-ietf-dnssd-srp-09#section-2.3.2) states:
```
   A DNS Update that contains any additional adds or deletes that cannot
   be identified as Service Discovery, Service Description or Host
   Description instructions is not an SRP update.  A DNS update that
   contains any prerequisites is not an SRP update.  Such messages
   should either be processed as regular RFC2136 updates, including
   access control checks and constraint checks, if supported, or else
   rejected with RCODE=REFUSED.
```
So the same server may handle both SRP updates and regular RFC2136 updates.

@Abhayakara, thoughts on the SRP server UDP port?

Reply:
Thread Technical Committee consensus is to use port 53 for SRP.